website/blog/2021-08-05-win32component.md

```diff
@@ -0,0 +1,416 @@
+authorURL: https://twitter.com/qmatteoq
+description: "How to use Win32 features from a React Native for Windows application"
+is_blog: true
+published_date: August 5, 2021
+---
+
+If you have adopted React Native to build your Windows applications, you'll know that the final output is a [Universal Windows Platform application](https://docs.microsoft.com/windows/uwp/get-started/universal-application-platform-guide). This development platform gives you access to all the latest enhancements in the Windows ecosystem (modern UI platform, notifications, integration with features like inking and Windows Hello, etc.), plus greater security and reliability thanks to the sandbox the application runs in.
+However, there might be scenarios where UWP isn't enough and you need to perform one or more tasks which are supported only by the Win32 ecosystem: working with any file on the disk without user intervention, reading a key from the registry, integrating a SDK which doesn't support the Windows Runtime.
+
+In this post we're going to explore a solution that will enable you to get the best of both worlds: a React Native for Windows application which integrates a classic Windows process. We're going to build a sample React Native application which will be able to read a registry key and display it. The goal is to display to the user values stored in the `\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion` hive, which contains many information about the currently installed version of Windows.
+
+<!--truncate-->
+
+The solution will use the following components.
+
+#### A classic Windows process
+
+We're going to build a .NET 5.0 console application which, by targeting Windows 10 and using the [Windows Compatibility Pack](https://docs.microsoft.com/dotnet/core/porting/windows-compat-pack), will be able to interact with the system registry. The React Native application will send to this process the name of the key we want to retrieve; the process will read its value and send it back to the React Native application.
+
+#### A React Native for Windows application
+
+This will be a traditional React Native for Windows application. However, as we're going to see later, we'll have to make a couple of changes to the host app generated by the template to handle the communication with the classic Windows process.
+
+#### App Services
+
+[App Services](https://docs.microsoft.com/windows/uwp/launch-resume/app-services) is a UWP feature that enables a Windows application to host one or more background tasks that can be consumed by other applications. You can think of this feature like REST services, but exposed locally. Once a UWP application which exposes an App Service is deployed on a machine, other applications can connect to it using its name and the Package Family Name of the hosting application. Once the communication channel is established, the calling application can send data to the App Service, which will process it and send back the results.
```

Comment:
other applications => other applications in the same package? (pls double check my understanding is correct)